### PR TITLE
ttd bid adapter: pass on all of user/app/ortb2imp for first party data support, default imp.secure

### PR DESCRIPTION
**Pull request by khang-vu-ttd**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Pass on all members in user/app/site for first party support
Default imp.secure  = 1 if not set

## Other information

@julian-burger-ttd

### CODE REVIEW

**Comment by patmmccann** [comment]
why not take pmp?

**Review by patmmccann — CHANGES_REQUESTED** [review]
Why not take PMP and device object also?

File: modules/ttdBidAdapter.js [hunk 1]
```diff
@@ -413,9 +404,10 @@ export const spec = {
     let topLevel = {
       id: bidderRequest.auctionId,
       imp: validBidRequests.map(bidRequest => getImpression(bidRequest)),
+      app: firstPartyData.app || {},
```
[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/ttdBidAdapter.js:407]: also this ends up making your request invalid potentially, site and app and dooh cannot cooccur

File: modules/ttdBidAdapter.js [hunk 2]
```diff
@@ -413,9 +404,10 @@ export const spec = {
...
       site: getSite(bidderRequest, firstPartyData),
       device: getDevice(),
-      user: getUser(bidderRequest),
+      user: getUser(bidderRequest, firstPartyData),
```
Comment (patmmccann) [inline modules/ttdBidAdapter.js:410]: why not take fpd on device?

Comment (khang-vu-ttd) [reply modules/ttdBidAdapter.js:410]: thanks for pointing device and pmp out, we don't expect to need them right now but its good to pass those along

File: modules/ttdBidAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (khang-vu-ttd) [reply modules/ttdBidAdapter.js:407]: just double checking is the first comment suppose to read 'why not pmp: firstPartyData.pmp || {}' ?

[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/ttdBidAdapter.js]: Yes

File: modules/ttdBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/ttdBidAdapter.js:411]: Okay

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
Thanks!!

**Pull request merged**